tests/ssl-opt.sh

```diff
@@ -1893,6 +1893,22 @@ run_test    "Authentication: client no cert, ssl3" \
             -C "! mbedtls_ssl_handshake returned" \
             -S "X509 - Certificate verification failed"
 
+# Tests for CA list suppression
+
+run_test    "Authentication: CA suppression disabled (default)" \
+            "$P_SRV debug_level=3 auth_mode=required" \
+            "$P_CLI crt_file=data_files/server6.crt \
+             key_file=data_files/server6.key" \
+            0 \
+            -s "requested DN"
+
+run_test    "Authentication: CA suppression enabled" \
+            "$P_SRV debug_level=3 auth_mode=required suppress_ca=1" \
+            "$P_CLI crt_file=data_files/server6.crt \
+             key_file=data_files/server6.key" \
+            0 \
+            -S "requested DN"
+
```

Comment:
IIRC currently a client can only have 0 or 1 certificate provisioned, not a longer list, so our client implementation currently just ignores the CA list and sends its certificate iff it has one. So, while I agree about adding a test that the client sends a cert when the server CA list is empty, I think the other test will show that our client indeed can send a cert for non-advertised CA.

The RFC doesn't seem to explicitly state what a client is supposed to do when it only has a certificate that doesn't match the non-empty list. But I remember discussions on the TLS mailing list about similar issues (no certificate matching signature_algorithms) and it seemed to me there was a consensus that sending the cert anyway was OK: the peer has to be prepared for than anyway, and is then free to react however it likes (accept the cert anyway (unlikely), continue the handshake considering the peer as _not_ authenticated, abort the handshake).

While the last test (client sends a non-trusted cert) is not about new behavior introduced by this PR, I'm all in favor of taking that occasion to include it if we didn't test that case already (I think/hope we do, but let's make sure).